src/context/WalletProvider/NativeWallet/NativeSuccess/hooks/useNativeSuccess/useNativeSuccess.ts

```diff
@@ -0,0 +1,46 @@
+import { NativeAdapter } from '@shapeshiftoss/hdwallet-native'
+import { EncryptedWallet } from '@shapeshiftoss/hdwallet-native/dist/crypto'
+import { useEffect, useState } from 'react'
+import { SUPPORTED_WALLETS } from 'context/WalletProvider/config'
+import { useWallet, WalletActions } from 'context/WalletProvider/WalletProvider'
+import { useLocalStorage } from 'hooks/useLocalStorage/useLocalStorage'
+
+export type UseNativeSuccessPropTypes = { encryptedWallet?: EncryptedWallet }
+
+export const useNativeSuccess = ({ encryptedWallet }: UseNativeSuccessPropTypes) => {
+  const [isSuccessful, setIsSuccessful] = useState<boolean | null>(null)
+  const [, setLocalStorageWallet] = useLocalStorage<Record<string, string>>('wallet', null)
+  const { state, dispatch } = useWallet()
+
+  useEffect(() => {
+    ;(async () => {
+      if (encryptedWallet?.encryptedWallet && state.adapters?.native) {
+        try {
+          let mnemonic = await encryptedWallet.decrypt()
+          const wallet = await (state.adapters.native as NativeAdapter).pairDevice(
+            encryptedWallet.deviceId
+          )
+          await wallet?.loadDevice({ mnemonic })
+          mnemonic = '' // Clear out the mnemonic as soon as we're done with it
+          setLocalStorageWallet({
+            [encryptedWallet.deviceId]: encryptedWallet.encryptedWallet
+          })
+          setIsSuccessful(true)
+          const { name, icon } = SUPPORTED_WALLETS['native']
+          dispatch({
+            type: WalletActions.SET_WALLET,
+            payload: { wallet, name, icon, deviceId: encryptedWallet.deviceId }
+          })
```

Comment:
I dont know if this matters but I was wondering, is `isSuccessful` just saying that it successfully got set in localStorage? What if something happened during the dispatch of `SET_WALLET` could the user be in a weird state where it said they successfully connected with the wallet in localStorage but it is not in the wallet context? 

So I was wondering if it made sense to move `setIsSuccessful(true)` to the bottom of the try so that we know all the things happened before we say it is a success.
```suggestion
          const { name, icon } = SUPPORTED_WALLETS['native']
          dispatch({
            type: WalletActions.SET_WALLET,
            payload: { wallet, name, icon, deviceId: encryptedWallet.deviceId }
          })
          setIsSuccessful(true)
```

Reply:
Yes good call.